src/TestFramework/TestFramework/Interfaces/ITestDataSourceIgnoreCapability.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+/// <summary>
+/// Specifies the capability of a test data source to be ignored and define the ignore reason.
+/// </summary>
+public interface ITestDataSourceIgnoreCapability
```

Comment:
Could it be worth making this one more generic (not linked to data sources) so we can apply it to test methods and test classes too?

Reply:
@Evangelink The impl is quite different, at least currently. For example, for specific test cases, we need to have a VSTest property for that to restore the associated ignore message. But we don't need that for when it's the whole test method or test class being ignored.